collector/collector.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	"github.com/google/cadvisor/info/v2"
+)
+
+type Collector struct {
+	//name of the collector
+	name string
+
+	//holds information extracted from the config file for a collector
+	configFile Config
+
+	//time at which metrics need to be collected next
+	nextCollectionTime time.Time
+
+	//error (if any) during metrics collection
+	err error
+}
+
+type Config struct {
+	//the endpoint to hit to scrape metrics
+	Endpoint string `json:"endpoint"`
+
+	//holds information about different metrics that can be collected
+	MetricsConfig []metricConfig `json:"metricsConfig"`
+}
+
+// metricConfig holds information extracted from the config file about a metric
+type metricConfig struct {
+	//the name of the metric
+	Name string `json:"name"`
+
+	//enum type for the metric type
+	MetricType MetricType `json:"metricType"`
+
+	//data type of the metric (eg: integer, string)
+	Units string `json:"units"`
+
+	//the frequency at which the metric should be collected (in seconds)
+	PollingFrequency string `json:"pollingFrequency"`
```

Comment:
We should represent this as a `time.Duration` or an integer and say it is in seconds.